lib/perl/Genome/Model/Tools/DetectVariants2/Dispatcher.pm

```diff
@@ -300,18 +303,25 @@ sub _dump_dv_cmd {
     return 1;
 }
 
-sub get_relative_path_to_output_directory {
+sub get_relative_path_to_result_directory {
     my $self = shift;
-    my $full_path = shift;
-    my $relative_path = $full_path;
-    my $temp_path = $self->_temp_staging_directory;
-    $relative_path =~ s/$temp_path\/?//;
-    return $relative_path;
+    my $result_path = shift;
+
+    my $relative_path = $result_path;
+
+    my $dispatcher_path = $self->output_directory;
+    if ($relative_path =~ /$dispatcher_path\/?/ ) {
+        $relative_path =~ s/$dispatcher_path\/?//;
+        return $relative_path;
+    }
+
+    return;
```

Comment:
Maybe this would be a good place to use `Path::Class`.  It has `contains` (or `subsumes`) and then we could use `$file->relative($dir)` to get the relative path.